app/modules/utils.py

```diff
@@ -22,3 +26,46 @@ def fail_on_missing_static_folder(
         raise RuntimeError(
             f'static folder improperly configured - could not locate a valid installation at: {folder}'
         )
+
+
+# Many module resources create things and then need to clean up when it fails. This helper class cleans up
+class Cleanup(object):
+    def __init__(self, name):
+        self.name = name
+        # For things where a guid has been created on EDM but not an object in Houston
```

Comment:
Is this likely? We ensure a Houston object is created before moving on to EDM creation I think.

Reply:
Not in the example of a sighting creation in both the way that Jon had it and the way that is implemented for the commit of the assetgroupsighting. 
The req is sent to edm. Only once we are sure that the resp properly matches the req do we create the appropriate houston objects.